### PR TITLE
Extract `fn read_dir_to_string_vec` for FUSE tests

### PR DESCRIPTION
**Pull request by dannycjones**
Extracting some common logic repeated all over the FUSE tests. Feedback from #232.

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and I agree to the terms of the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).

### CODE REVIEW

**Comment by dannycjones** [comment]
CI is failing ASAN because of something breaking with Nightly Rust. I'd propose we ignore - though we may need to investigate why our Nightly Rust ASAN tests are failing.

File: mountpoint-s3/tests/fuse_tests/mod.rs [hunk 1]
```diff
@@ -222,3 +225,17 @@ mod s3_session {
         }
     }
 }
+
+/// Take a `read_dir` iterator and return the entry names
+pub fn read_dir_to_entry_names(read_dir_iter: ReadDir) -> Vec<String> {
+    read_dir_iter
+        .map(|entry_result| entry_result.expect("no io err during readdir"))
```
Comment (passaro) [inline mountpoint-s3/tests/fuse_tests/mod.rs:232]: nit: is there a reason to have multiple `map`s here?

**Review by passaro — COMMENTED** [review]
LGTM

File: mountpoint-s3/tests/fuse_tests/mod.rs (hunk 1 above)
Comment (dannycjones) [reply mountpoint-s3/tests/fuse_tests/mod.rs:232]: It doesn't matter either way - only to break into logical steps which we can do with local variables as well.

Happy to change if you would find that more readable.

Comment (dannycjones) [reply mountpoint-s3/tests/fuse_tests/mod.rs:232]: I'll push it anyway as a single map

**Review by passaro — APPROVED** [review]

**Pull request merged**